api/src/main/java/org/eclipse/microprofile/metrics/MetricRegistry.java

```diff
@@ -109,7 +109,7 @@ private static void append(StringBuilder builder, String part) {
      * @return {@code klass} and {@code names} concatenated by periods
      */
     public static String name(Class<?> klass, String... names) {
-        return name(klass.getName(), names);
+        return name(klass.getCanonicalName(), names);
```

Comment:
I'm not yet completely sure about this, but I suppose canonical name would be better (in case of a nested class, the short name of the nested class will be separated by `.` instead of `$`). The suggested spec text now also specifies that it must be the canonical name (I think in 2.3 it's not clear at all from the spec what name should be used)

Reply:
looking at https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav it seems like the canonical form is nicer for the simpler cases, but wouldn't have a value for some situations, such as local classes...

```
getName():          com.example.TestClassNames$1LocalClass
getCanonicalName(): null
getSimpleName():    LocalClass
toString():         class com.example.TestClassNames$1LocalClass
```

Reply:
I don't think local classes make any sense to be used for Metrics. Or with CDI. CDI spec says that inner classes aren't managed beans, and AFAIK local classes count as inner classes.
So I'd say we can live with that?

Reply:
thanks @jmartisk , that's a good point.